script/import_scripts/ipboard.rb

```diff
@@ -0,0 +1,1019 @@
+        # new_text += "type: #{content_type}<br>\n"
+        if content_class.length > 0 && content_class != "forums_Topic"
+          new_text += "UNEXPECTED CONTENT CLASS! #{content_class}<br>\n"
+        end
+        if content_type.length > 0 && content_type != "forums"
+          new_text += "UNEXPECTED CONTENT TYPE! #{content_type}<br>\n"
+        end
+        # puts "#{'-'*20} and NOWWWWW!!!! \n #{new_text}"
+      end
+      if post && topic && username
+        quote = "\n[quote=\"#{username}, post:#{post}, topic: #{topic}\"]\n\n"
+      else
+        if username && username.length > 1
+          quote = "\n[quote=\"#{username}\"]\n\n"
+        else
+          quote = "\n[quote]\n"
+        end
+        # new_doc = Nokogiri::XML("<div>#{new_text}</div>")
+      end
+      puts "QUOTE: #{quote}"
+      sleep 1
+      b.content = quote + b.content + "\n[/quote]\n"
+      b.name = 'div'
+    end
+
+    raw = @doc.to_html
+  end
+
+  def clean_up(raw)
+    return "" if raw.blank?
+
+    raw.gsub!(/<#EMO_DIR#>/, EMO_DIR)
+    # TODO what about uploads?
+    # raw.gsub!(/<fileStore.core_Attachment>/,UPLOADS)
+    raw.gsub!(/<br>/, "\n\n")
+    raw.gsub!(/<br \/>/, "\n\n")
+    raw.gsub!(/<p>&nbsp;<\/p>/, "\n\n")
+     raw.gsub!(/\[hr\]/,"\n***\n")
+    raw.gsub!(/&#39;/, "'")
+    raw.gsub!(/\[url="(.+?)"\]http.+?\[\/url\]/, "\\1\n")
+    raw.gsub!(/\[media\](.+?)\[\/media\]/, "\n\\1\n\n")
+    raw.gsub!(/\[\/quote\]/, "\n[/quote]\n")
+    raw.gsub!(/date=\'(.+?)\'/, '')
+    raw.gsub!(/timestamp=\'(.+?)\' /, '')
+
+    quote_regex = /\[quote name=\'(.+?)\'\s+post=\'(\d+?)\'\s*\]/
+    while quote = quote_regex.match(raw)
+      # get IPB post number and find Discourse post and topic number
+      puts "----------------------------------------\nName: #{quote[1]}, post: #{quote[2]}" unless QUIET
+      imported_post_id = quote[2].to_s
+      topic_lookup = topic_lookup_from_imported_post_id("comment#" + imported_post_id)
+      post_lookup = topic_lookup_from_imported_post_id("discussion#" + imported_post_id)
+      puts "topic_lookup: #{topic_lookup}, post: #{post_lookup}" unless QUIET
+      post_num = topic_lookup ? topic_lookup[:post_number] : nil
+      topic_num = topic_lookup ? topic_lookup[:topic_id] : nil
+      post_num ||= post_lookup ? post_lookup[:post_number] : nil
+      topic_num ||= post_lookup ? post_lookup[:topic_id] : nil
+
+      # Fix or leave bogus username?
+      username = find_user_by_import_id(quote[1]) || quote[1]
+      puts "username: #{username}, post_id: #{post_num}, topic_id: #{topic_num}" unless QUIET
+      puts "Before fixing a quote: #{raw}\n**************************************** " unless QUIET
+      post_string = post_num ? ", post:#{post_num}" : ""
+      topic_string = topic_num ? ", topic:#{topic_num}" : ""
+      raw.gsub!(quote_regex, "\n[quote=\"#{username}#{post_string}#{topic_string}\"]\n\n")
+      puts "AFTER!!!!!!!!!!!!1: #{raw}" unless QUIET
+      sleep 1
```

Comment:
Why sleep here?

Reply:
I think it must have been because I was tired?

This is not code that I'm proud of, but I'm sick of it and people are asking for it, so I'm submitting bad code rather than no code. If this is the most embarrassing thing you find, I'll be elated.